src/sentry/api/endpoints/organization_discover.py

```diff
@@ -73,24 +93,33 @@ def validate_projects(self, attrs, source):
 
         return attrs
 
-    def validate_fields(self, attrs, source):
-        # If we're including exception_stacks.* or exception_frames.* fields
-        # then add arrayjoin value so this gets returned as strings
+    def validate_conditions(self, attrs, source):
+        # Handle exception_stacks, exception_frames
+        if attrs.get(source):
+            conditions = [self.get_condition(condition) for condition in attrs[source]]
+            attrs[source] = conditions
+        return attrs
+
+    def get_array_field(self, field):
         pattern = r"^(exception_stacks|exception_frames)\..+"
-        match = next(
-            (
-                re.search(pattern, field).group(1)
-                for field
-                in attrs.get(source, [])
-                if re.match(pattern, field)
-            ),
-            None
-        )
+        return re.search(pattern, field)
 
-        if match:
-            attrs['arrayjoin'] = match
+    def get_condition(self, condition):
+        array_field = self.get_array_field(condition[0])
+        has_equality_operator = condition[1] in ('=', '!=')
 
-        return attrs
+        # Apply has function to any array field if it's = / != and not part of arrayjoin
+        if array_field and has_equality_operator and (array_field.group(1) != self.arrayjoin):
+            value = condition[2]
+
+            if (isinstance(value, six.string_types)):
+                value = "'{}'".format(value)
```

Comment:
You could have a UnicodeError here if `value` has non-ascii bytes.

Reply:
good catch, thanks